### PR TITLE
Mixed bag clean up

### PR DESCRIPTION
**Pull request by calderjo**
yeah handful of things

1) Combined some layer

2) made comments referring to bugs use consistent format 

3) removed low usage packages and tests, we recently launched package manager. we can confidently remove more unused packages. The following packages are to be removed:
- pykalman
- preprocessing
- hmmlearn
- gplearn
- geoplot
- polyglot
- ggplot
- descartes
- fitter
- imagecodecs
- wfdb
- hpsklearn
- cleverhans
- osmnx
- pysal
- wordsegment
- vowpalwabbit
- feather
- kmodes
- ortools
- mlens
- vecstack
- Geohash
- geoviews
- s2sphere
- flashtext
- kmapper
- stemming
- hunspell
- spectral
- essentia
- hypertools
- stop_words
- scattertext
- vaex
- blake3
- catalyst (note no longer maintained since 2022)

### CODE REVIEW

File: tests/test_imports.py [hunk 1]
```diff
@@ -4,6 +4,4 @@ class TestImport(unittest.TestCase):
     # Basic import tests for packages without any.
     def test_basic(self):
         import bq_helper
-        import cleverhans
         import tensorflow_datasets
-        import segment_anything
```
Comment (djherbis) [inline tests/test_imports.py:9]: Did we drop segment_anything?

File: tests/test_matplotlib.py [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
 
+from distutils.version import StrictVersion
+
+import matplotlib
 import matplotlib.pyplot as plt
 import numpy as np
 
 class TestMatplotlib(unittest.TestCase):
+    def test_version(self):
+        self.assertLess(StrictVersion(matplotlib.__version__), StrictVersion("3.8.0"))
```
Comment (djherbis) [inline tests/test_matplotlib.py:12]: Can we comment why we need this less than this version?

**Review by djherbis — APPROVED** [review]

File: tests/test_imports.py (hunk 1 above)
Comment (calderjo) [reply tests/test_imports.py:9]: my bad, i may have mixed it up with another package. usage seems decent to keep atm

File: tests/test_matplotlib.py (hunk 2 above)
Comment (calderjo) [reply tests/test_matplotlib.py:13]: d.

**Pull request merged**